src/app/core/identity-provider/identity-provider.interface.ts

```diff
@@ -1,9 +1,17 @@
 import { HttpEvent, HttpHandler, HttpRequest } from '@angular/common/http';
 import { ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
-import { Observable } from 'rxjs';
+import { Observable, isObservable, of } from 'rxjs';
+
+import { isPromise } from 'ish-core/utils/functions';
 
 export type TriggerReturnType = Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree;
 
+export function getTriggerReturnType$(
```

Comment:
this method should be moved - it should not be part of an interface file, pls also add documentation